platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/Collector.java

```diff
@@ -20,7 +20,10 @@
 import org.gradle.api.Action;
 
 /**
- * A supplier of zero or more values of type {@link T}.
+ * A collector is a value supplier of zero or more values of type {@link T}.
+ * <p>
+ *     A <code>Collector</code> represents an increment to a collection property.
```

Comment:
💅 I think this line should say `can represent`, as there's no reason this interface couldn't be used in other ways.

Reply:
The thing is that `Collector` (and `CollectionSupplier`) are really like implementation-oriented sealed class hierarchies,  which only exist for the purpose the design needs them, they are not meant to be used in other contexts.